File: 1448-count-good-nodes-in-binary-tree/1448-count-good-nodes-in-binary-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int goodNodes(TreeNode* root) {
        return helper(root, INT_MIN);
    }
    
    int helper(TreeNode* root, int maxV){
        if (!root) return 0;
        int ans{0};
        if(root->val >= maxV){
            maxV = root->val;
            ++ans;
        }
        return helper(root->left, maxV) + helper(root->right, maxV) + ans;
    }
};

File: 1448-count-good-nodes-in-binary-tree/1448-count-good-nodes-in-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def goodNodes(self, root: TreeNode) -> int:
        self.ans = 0
        
        def dfs(maxV, node):
            if not node:
                return
            
            if maxV <= node.val:
                self.ans += 1
            
            maxV = max(node.val, maxV)
            dfs(maxV, node.left)
            dfs(maxV, node.right)
        
        dfs(float('-inf'), root)
        return self.ans
                
            
            